picard/session/metadata_handler.py

```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2025 The MusicBrainz Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+"""Metadata handling for session management.
+
+This module provides utilities for serializing and deserializing metadata
+for session files, with proper error handling and validation.
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+
+from picard import log
+from picard.file import File
+from picard.metadata import Metadata
+from picard.session.constants import SessionConstants
+
+
+class MetadataHandler:
+    """Handles metadata serialization and deserialization for sessions."""
+
+    @staticmethod
+    def serialize_metadata_for_file(file: File) -> dict[str, list[Any]]:
+        """Serialize metadata for a file, excluding internal tags.
+
+        Parameters
+        ----------
+        file : File
+            The file to serialize metadata for.
+
+        Returns
+        -------
+        dict[str, list[Any]]
+            Dictionary containing serialized metadata tags.
+
+        Notes
+        -----
+        Only user-visible tags are serialized, internal tags (starting with ~)
+        and length are excluded.
+        """
+        tags: dict[str, list[Any]] = {}
+        for key, values in file.metadata.rawitems():
+            if key.startswith(SessionConstants.INTERNAL_TAG_PREFIX) or key == "length":
```

Comment:
The special case of `length` is spreading too much IMHO, we should have something more generic, in case we need to add a similar special case later.
At least something like `or key in SPECIAL_TAGS` (random name, could be better).